### DISCUSSION TITLE
Capture ANSI output from stdout / disabling "dumb" terminal mode?

### DISCUSSION THREAD

**Discussion by bmeares** [Textualize/rich#1191] [Q&A] [answered]
Hi there! First of all, absolutely loving `rich`. I am looking for ways to capture the ANSI characters produced by `rich` that are printed to stdout. I have tested and am able to capture ANSI with frameworks like [`more_termcolor.colored`](https://github.com/giladbarnea/more_termcolor/blob/8d1cce7f033a77559a83388cff3ff9dc06bf0d35/more_termcolor/main.py#L485), but it seems that the "dumb" terminal protection offered by `rich` strips ANSI characters if text isn't being printed to a tty.

For context, I am capturing the output from subprocesses and rendering them on a webpage with [`ansi_up.js`](https://github.com/drudru/ansi_up).

**Comment by willmcgugan** [comment, marked as answer]
You can add `force_terminal=True` and `color_system="truecolor"`  to the Console constructor.

BTW Rich can [export](https://rich.readthedocs.io/en/latest/console.html#exporting) directly to HTML.

**Reply by bmeares** [reply]
Thanks so much! I wasn't directly creating a Console object so I missed those options. Also the capture functionality of `console.capture()` looks super useful! `rich` really has everything.
